src/main/java/org/javarosa/core/model/instance/geojson/GeojsonFeature.java

```diff
@@ -28,6 +28,8 @@
     private GeojsonGeometry geometry;
     private Map<String, String> properties;
 
+    private String id;
```

Comment:
What's the goal of parsing this out? Is it so it can be referenced in forms (questions/choice filters etc)?

Reply:
It's to support a top-level `id` because that has meaning in the geojson spec. Other top-level properties are ignored (per that dreadful commit). Commit details:

> From GeoJSON spec: "If a Feature has a commonly used identifier, that identifier SHOULD be included as a member of the Feature object with the name "id", and the value of this member is either a JSON string or number." [https://datatracker.ietf.org/doc/html/rfc7946\#section-3.2](https://datatracker.ietf.org/doc/html/rfc7946/#section-3.2)

We give priority to the toplevel id since it's part of the spec but fall back to `properties/id` since it's also commonly used (see https://github.com/qgis/QGIS/issues/40876, for example).

The `id` node in JavaRosa's internal representation (which could come from top level or `properties/id`) is going to be the default underlying value for selects: https://github.com/XLSForm/pyxform/issues/591 But yeah, you could also use it arbitrarily in a form.